lib/java/AdapterLibrary/src/main/kotlin/com/vmware/aria/operations/Result.kt

```diff
@@ -0,0 +1,412 @@
+
+    /**
+     * Returns all [Objects][Object] with the given type.
+     * @param objectType The [object type][Key.objectType] to match
+     * @return A [List] of [Objects][Object] matching the [objectType]
+     */
+    fun getObjectsByType(objectType: String) =
+        objects.values.filter { obj ->
+            obj.objectType == objectType
+        }
+
+    /**
+     * Returns all [Objects][Object] with the given adapter type and object type.
+     * @param adapterType The [adapter type][Key.adapterType] to match
+     * @param objectType The [object type][Key.objectType] to match
+     * @return A [List] of [Objects][Object] matching the [adapterType] and [objectType]
+     */
+    fun getObjectsByType(adapterType: String, objectType: String) =
+        objects.values.filter { obj ->
+            obj.adapterType == adapterType && obj.objectType == objectType
+        }
+
+    /**
+     * Returns all [Objects][Object] with the given adapter type.
+     * @param adapterType The [adapter type][Key.adapterType] to match
+     * @return A [List] of [Objects][Object] matching the [adapterType] and [objectType]
+     */
+    fun getObjectsByAdapterType(adapterType: String) =
+        objects.values.filter { obj ->
+            obj.adapterType == adapterType
+        }
+
+    /**
+     * Adds the given [Object] to the [CollectResult] and returns it.
+     *
+     * A different [Object] with the same key cannot already exist in the [CollectResult].
+     * If it does, an [ObjectKeyAlreadyExistsException] will be thrown.
+     *
+     * @param obj The [Object] to add to the [CollectResult].
+     * @return The object.
+     * @throws ObjectKeyAlreadyExistsException: If a different [Object] with the same [Key]
+     * already exists in the [CollectResult].
+     */
+    fun addObject(obj: Object): Object {
+        val o = objects.getOrPut(obj.key) { obj }
+        if (o === obj)
+            return o
+        throw ObjectKeyAlreadyExistsException(obj.key)
+    }
+
+    /**
+     * Adds the given [Objects][Object] to the [CollectResult].
+     *
+     * A different [Object] with the same key cannot already exist in the [CollectResult].
+     * If it does, an [ObjectKeyAlreadyExistsException] will be thrown. All [Objects][Object]
+     * will attempt to add to the [CollectResult].
+     *
+     * @param objects A [Collection] of [Objects][Object] to add to the [CollectResult].
+     * @throws ObjectKeyAlreadyExistsException: If any [Objects][Object] with the same [Key]
+     * already exists in the [CollectResult].
+     */
+    fun addObjects(objects: Iterable<Object>) {
+        val failed = mutableListOf<Key>()
+        for (obj in objects) {
+            try {
+                addObject(obj)
+            } catch (_: ObjectKeyAlreadyExistsException) {
+                failed.add(obj.key)
+            }
+        }
+        if (failed.isNotEmpty()) {
+            throw ObjectKeyAlreadyExistsException(failed)
+        }
```

Comment:
Here's the only place where we're calling the exception with the list constructor